remoto/connection.py

```diff
@@ -21,9 +21,16 @@ def __init__(self, hostname, logger=None, sudo=False, threads=1, eager=True,
         self.interpreter = interpreter or 'python%s' % sys.version_info[0]
 
         if eager:
-            if detect_sudo:
-                self.sudo = self._detect_sudo()
-            self.gateway = self._make_gateway(hostname)
+            try:
+                if detect_sudo:
+                    self.sudo = self._detect_sudo()
+                self.gateway = self._make_gateway(hostname)
+            except OSError:
```

Comment:
which one might raise the `OSError`? is it `self._detect_sudo()` or `self._make_gateway(hostname)` ?

Ideally this try/except would happen within that call, not here in the `__init__`